packages/console/src/components/ActionsButton/index.tsx

```diff
@@ -6,30 +6,55 @@ import Delete from '@/assets/icons/delete.svg';
 import Edit from '@/assets/icons/edit.svg';
 import More from '@/assets/icons/more.svg';
 import ActionMenu, { ActionMenuItem } from '@/ds-components/ActionMenu';
+import type { Props as ButtonProps } from '@/ds-components/Button';
 import ConfirmModal from '@/ds-components/ConfirmModal';
 import DynamicT from '@/ds-components/DynamicT';
 import useActionTranslation from '@/hooks/use-action-translation';
 
 import * as styles from './index.module.scss';
 
 type Props = {
+  /**
+   * Props that will be passed to the button that opens the menu. It will override the
+   * default props.
+   */
+  buttonProps?: Partial<ButtonProps>;
   /** A function that will be called when the user confirms the deletion. */
   onDelete: () => void | Promise<void>;
-  /** A function that will be called when the user clicks the edit button. */
-  onEdit: () => void | Promise<void>;
+  /**
+   * A function that will be called when the user clicks the edit button. If not provided,
+   * the edit button will not be displayed.
+   */
+  onEdit?: () => void | Promise<void>;
```

Comment:
Seems you can only edit or delete in this component. But there are more other usages in our console app. How about passing the action menu items in an array of objects, like [this](https://github.com/logto-io/logto/pull/4712/files#diff-c147b7fba4a86c2515199c7dbcb16491fc854344bb2c943bbc64a8763d6cadb6R223-R232)?

Reply:
don't want to over engineering until it's really necessary

Reply:
It is necessary in my PR (#4712), though. And I'm afraid it would have a conflict later with your follow-up PRs

Reply:
sure np